middlewares/auth.js

```diff
@@ -0,0 +1,27 @@
+const db = require("../models");
+
+async function isEmailValidDB(req, res, next) {
+  try {
+    const email = req.body.email;
+    const user = await db.User.findOne({
+      where: {
+        email,
+      },
+    });
+    if (user) {
+      return res.status(200).json({
+        message: "Email already in use",
+      });
+    }
+    next();
+  } catch (error) {
+    res.status(404).json({
+      message: "Error validating email",
+      error,
+    });
+  }
```

Comment:
Habria que ver el motivo del error, si falla la conexion con la base de datos por ejemplo, no se deberia devolver un 404 (not-found) sino un 500, ya que el error es interno al servidor.